File: Swift/BOJ/1000/1026.swift
//
//  1026.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/05.
//

import Foundation

let n = Int(readLine()!)!
let listA = readLine()!.split { $0 == " " }.map { Int(String($0))! }.sorted()
let listB = readLine()!.split { $0 == " " }.map { Int(String($0))! }.sorted(by: >)

var result = 0
for i in 0..<listA.count {
    result += listA[i] * listB[i]
}

print(result)

File: Python/BOJ/그리디/1026.py
//
//  1026.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/05.
//

import sys; readLine = sys.stdin.readline

n = int(readLine())
a = list(map(int, readLine().split()))
b = list(map(int, readLine().split()))

a.sort()
b.sort(reverse=True)

result = 0
for i in range(len(a)):
  result += a[i] * b[i]

print(result)
